Atlas panics when trying to generate migrations

When running `make generate` in controlplane service, atlas panics with this error:
```
> make generate
cd ./plugins/sdk/v1/plugin/api && buf generate
cd ./api && buf generate
cd ./internal/conf && buf generate
atlas migrate hash --dir file://internal/data/ent/migrate/migrations
atlas migrate diff --dir file://internal/data/ent/migrate/migrations --to "ent://internal/data/ent/schema" --dev-url "docker://postgres/15/test?search_path=public"
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x104740224]

goroutine 106 [running]:
go/types.(*Checker).handleBailout(0x1400027e800, 0x14000d6db98)
	/opt/homebrew/Cellar/go/1.22.2/libexec/src/go/types/check.go:367 +0x9c
panic({0x105c4f0c0?, 0x106badc80?})
	/opt/homebrew/Cellar/go/1.22.2/libexec/src/runtime/panic.go:770 +0x124
go/types.(*StdSizes).Sizeof(0x0, {0x105e9a028, 0x106bebba0})
	/opt/homebrew/Cellar/go/1.22.2/libexec/src/go/types/sizes.go:228 +0x314
go/types.(*Config).sizeof(...)
	/opt/homebrew/Cellar/go/1.22.2/libexec/src/go/types/sizes.go:333
go/types.representableConst.func1({0x105e9a028?, 0x106bebba0?})
	/opt/homebrew/Cellar/go/1.22.2/libexec/src/go/types/const.go:76 +0x9c
go/types.representableConst({0x105ea2798, 0x1070e0320}, 0x1400027e800, 0x106bebba0, 0x14000d6c838)
	/opt/homebrew/Cellar/go/1.22.2/libexec/src/go/types/const.go:106 +0x2b0
go/types.(*Checker).representation(0x1400027e800, 0x14001e9c340, 0x106bebba0)
	/opt/homebrew/Cellar/go/1.22.2/libexec/src/go/types/const.go:256 +0x68
go/types.(*Checker).representable(0x1400027e800, 0x14001e9c340, 0x106bebba0)
	/opt/homebrew/Cellar/go/1.22.2/libexec/src/go/types/const.go:239 +0x28
go/types.(*Checker).shift(0x1400027e800, 0x14001e9c300, 0x14001e9c340, {0x105e9f520, 0x14000d38f30}, 0x14)
	/opt/homebrew/Cellar/go/1.22.2/libexec/src/go/types/expr.go:650 +0x1d8
go/types.(*Checker).binary(0x1400027e800, 0x14001e9c300, {0x105e9f520, 0x14000d38f30}, {0x105e9fa90, 0x140006960e0}, {0x105e9f280, 0x14000696120}, 0x14, 0xb43d0f)
	/opt/homebrew/Cellar/go/1.22.2/libexec/src/go/types/expr.go:796 +0x100
go/types.(*Checker).exprInternal(0x1400027e800, 0x0, 0x14001e9c300, {0x105e9f520, 0x14000d38f30}, {0x0, 0x0})
	/opt/homebrew/Cellar/go/1.22.2/libexec/src/go/types/expr.go:1416 +0x1d4
go/types.(*Checker).rawExpr(0x1400027e800, 0x0, 0x14001e9c300, {0x105e9f520?, 0x14000d38f30?}, {0x0?, 0x0?}, 0x0)
	/opt/homebrew/Cellar/go/1.22.2/libexec/src/go/types/expr.go:979 +0x12c
go/types.(*Checker).expr(0x1400027e800, 0x14000c30540?, 0x14001e9c300, {0x105e9f520?, 0x14000d38f30?})
	/opt/homebrew/Cellar/go/1.22.2/libexec/src/go/types/expr.go:1513 +0x38
go/types.(*Checker).constDecl(0x1400027e800, 0x14000d4a360, {0x105e9f280, 0x140006960c0}, {0x105e9f520, 0x14000d38f30}, 0x0)
	/opt/homebrew/Cellar/go/1.22.2/libexec/src/go/types/decl.go:488 +0x23c
go/types.(*Checker).objDecl(0x1400027e800, {0x105ebede0, 0x14000d4a360}, 0x0)
	/opt/homebrew/Cellar/go/1.22.2/libexec/src/go/types/decl.go:191 +0x84c
go/types.(*Checker).packageObjects(0x1400027e800)
	/opt/homebrew/Cellar/go/1.22.2/libexec/src/go/types/resolver.go:693 +0x468
go/types.(*Checker).checkFiles(0x1400027e800, {0x14001da03f0, 0x2, 0x2})
	/opt/homebrew/Cellar/go/1.22.2/libexec/src/go/types/check.go:408 +0x164
go/types.(*Checker).Files(...)
	/opt/homebrew/Cellar/go/1.22.2/libexec/src/go/types/check.go:372
golang.org/x/tools/go/packages.(*loader).loadPackage(0x140000d60e0, 0x140005d1b00)
	/Users/jiparis/go/pkg/mod/golang.org/x/tools@v0.8.1-0.20230428195545-5283a0178901/go/packages/packages.go:1052 +0x870
golang.org/x/tools/go/packages.(*loader).loadRecursive.func1()
	/Users/jiparis/go/pkg/mod/golang.org/x/tools@v0.8.1-0.20230428195545-5283a0178901/go/packages/packages.go:851 +0x178
sync.(*Once).doSlow(0x0?, 0x0?)
	/opt/homebrew/Cellar/go/1.22.2/libexec/src/sync/once.go:74 +0x100
sync.(*Once).Do(...)
	/opt/homebrew/Cellar/go/1.22.2/libexec/src/sync/once.go:65
golang.org/x/tools/go/packages.(*loader).loadRecursive(0x0?, 0x0?)
	/Users/jiparis/go/pkg/mod/golang.org/x/tools@v0.8.1-0.20230428195545-5283a0178901/go/packages/packages.go:839 +0x50
golang.org/x/tools/go/packages.(*loader).refine.func2(0x0?)
	/Users/jiparis/go/pkg/mod/golang.org/x/tools@v0.8.1-0.20230428195545-5283a0178901/go/packages/packages.go:774 +0x30
created by golang.org/x/tools/go/packages.(*loader).refine in goroutine 1
	/Users/jiparis/go/pkg/mod/golang.org/x/tools@v0.8.1-0.20230428195545-5283a0178901/go/packages/packages.go:773 +0xaa4
make: *** [migration_sync] Error 2
```

I'm running it in MacOS 14.4.1 (M3).

--- a/common.mk
+++ b/common.mk
@@ -5,7 +5,9 @@ VERSION=$(shell git describe --tags --always)
 init: init-api-tools
 	go install github.com/google/wire/cmd/wire@latest
 	go install github.com/vektra/mockery/v2@v2.20.0
-	go install ariga.io/atlas/cmd/atlas@v0.12.0
+	# using binary release for atlas, since ent schema handler is not included
+	# in the community version anymore https://github.com/ariga/atlas/issues/2388#issuecomment-1864287189
+	curl -sSf https://atlasgo.sh | sh -s -- -y
 
 # initialize API tooling
 .PHONY: init-api-tools

